cocos/platform/CCFileUtils.h

```diff
@@ -41,6 +42,66 @@ NS_CC_BEGIN
  * @{
  */
 
+
+class ResizableBuffer {
+public:
+    virtual ~ResizableBuffer() {}
+    virtual void resize(size_t size) = 0;
+    virtual void* buffer() const = 0;
+};
+
+template<typename T>
+class ResizableBufferAdapter { };
+
+
+template<typename C, typename T, typename A>
+class ResizableBufferAdapter< std::basic_string<C, T, A> > : public ResizableBuffer {
```

Comment:
OK, I'll follow the name in here: http://en.cppreference.com/w/cpp/string/basic_string